src/ResourceManager/KeyVault/Commands.KeyVault/Models/KeyVaultManagementCmdletBase.cs

```diff
@@ -206,22 +206,35 @@ protected Guid GetObjectId(Guid objectId, string upn, string spn)
                     s.ServicePrincipalNames.Any(n => n.Equals(spn)))
                     .ExecuteAsync().GetAwaiter().GetResult().CurrentPage.FirstOrDefault();
                 if (servicePrincipal != null)
-                    objId = Guid.Parse(servicePrincipal.ObjectId);
+                    objId = servicePrincipal.ObjectId;
             }
-            else if (objectId != Guid.Empty)
+            else if (!string.IsNullOrWhiteSpace(objectId))
             {
-                var objectCollection = ActiveDirectoryClient.GetObjectsByObjectIdsAsync(new[] { objectId.ToString() }, new string[] { }).GetAwaiter().GetResult();
+                var objectCollection = ActiveDirectoryClient.GetObjectsByObjectIdsAsync(new[] { objectId }, new string[] { }).GetAwaiter().GetResult();
                 if (objectCollection.Any())
                     objId = objectId;
             }
 
-            if (objId != Guid.Empty)
+            if (!string.IsNullOrWhiteSpace(objId))
                 return objId;
 
             throw new ArgumentException(string.Format(PSKeyVaultProperties.Resources.ADObjectNotFound, objectFilter,
                 (_dataServiceCredential != null) ? _dataServiceCredential.TenantId : string.Empty));
         }
 
+        protected bool IsValidObjectId(string objectId)
+        {
+            // In ADFS, object IDs have no additional restrictions.
```

Comment:
There is an ongoing discussion on the SID serialized format, so this value may not be SID.ToString(). Thus not making any binding to the syntax of value for ADFS right now.